test/Misc/stats_dir_long_path_name.swift

```diff
@@ -0,0 +1,16 @@
+// REQUIRES: OS=macosx
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: mkdir -p %t/very-long-directory-name-that-contains-over-128-characters-which-is-not-enough-to-be-illegal-on-its-own-but-presents
+// RUN: cp %s %t/very-long-directory-name-that-contains-over-128-characters-which-is-not-enough-to-be-illegal-on-its-own-but-presents/a-problem-when-combined-with-other-long-path-elements-and-filenames-to-exceed-256-characters-combined-because-yay-arbitrary-limits-amirite.swift
+// RUN: touch %t/main.swift
+// RUN: %target-swiftc_driver -o %t/main -module-name main -stats-output-dir %t %t/main.swift %t/very-long-directory-name-that-contains-over-128-characters-which-is-not-enough-to-be-illegal-on-its-own-but-presents/a-problem-when-combined-with-other-long-path-elements-and-filenames-to-exceed-256-characters-combined-because-yay-arbitrary-limits-amirite.swift
```

Comment:
Why `%target-swiftc_driver` rather than `%target-build-swift`?

Reply:
I'm not clear on the difference! Does it matter here?

Reply:
I guess it probably doesn't. `%target-build-swift` includes things like "pass -O when testing optimized", and that's not so relevant for this test. I mostly asked because I just default to `%target-build-swift` and wondered why you didn't.